ion-schema/src/isl/isl_type_reference.rs

```diff
@@ -143,29 +149,82 @@ impl IslTypeRefImpl {
                     })?;
 
                 // check for nullable type reference
-                if value.annotations().any(|a| a.text() == Some("nullable")) {
-                    let built_in_type_def = IslTypeRefImpl::get_nullable_type_reference_definition(type_name)?;
-                    let value = &Element::read_one(built_in_type_def.as_bytes()).unwrap();
-                    return IslTypeRefImpl::from_ion_element(isl_version, value, inline_imported_types);
+                if value.has_annotation("nullable") {
+                    if isl_version == IslVersion::V2_0 {
+                        return invalid_schema_error(
+                            "`nullable::` modifier is not supported since ISL 2.0",
+                        )
+                    }
+
+                    return match type_name {
+                        // TODO: currently it only allows for built in types to be defined with `nullable` annotation for `document` and all other type references it returns an error
+                        "int" | "float" | "bool" | "decimal" | "string" | "symbol" | "blob" | "clob" | "timestamp" | "struct" | "sexp" | "list" | "text" | "lob" | "number" | "any" | "nothing" => {
```

Comment:
Yes, as you mentioned both `$ION_TYPE` and `ION_TYPE` will actually be identical in terms of validation because we have an explicit check for `IonType::Null`.

Reply:
> Yes, as you mentioned both $ION_TYPE and ION_TYPE will actually be identical in terms of validation because we have an explicit check for IonType::Null.

Right, but it's not a question of the validation. It's a matter of whether `nullable::$null` is _legal_ in Ion Schema 1.0. If you have `nullable::$null` should that result in an invalid schema error?

Reply:
Yes. I meant we should support all the `$ION_TYPES` because AFAIK we haven't specified these to be illegal anywhere in the Specification which would make people expect it as legal.